velox/exec/tests/ContainerRowSerdeTest.cpp

```diff
@@ -144,5 +144,36 @@ TEST_F(ContainerRowSerdeTest, nested) {
   testRoundTrip(nestedArray);
 }
 
+TEST_F(ContainerRowSerdeTest, compareDoubleEquals) {
+  static constexpr CompareFlags flags{true, true, false, false};
+  auto data = makeFlatVector<double>(
+      {std::numeric_limits<double>::quiet_NaN(), 2.0, 3.0});
+
+  // Test byte stream and array input.
+  ByteStream left;
+  auto leftPosition = serialize(data);
+  HashStringAllocator::prepareRead(leftPosition.header, left);
+
+  SelectivityVector allRows(data->size());
```

Comment:
Do we need this DecodedVector ? Can we not compare with a FlatVector ?

Reply:
Yes. The API requires a DecodedVector (there is a difference in the interface and implementation). When using FlatVector (which I originally tried) compilation fails with
```
/Users/czentgr/gitspace/velox/velox/exec/tests/ContainerRowSerdeTest.cpp:160:18: error: no matching function for call to 'compare'
    EXPECT_EQ(0, ContainerRowSerde::compare(left, data, i, flags));
...
/Users/czentgr/gitspace/velox/./velox/exec/ContainerRowSerde.h:35:18: note: candidate function not viable: no known conversion from 'FlatVectorPtr<EvalType<double>>' (aka 'shared_ptr<FlatVector<double>>') to 'const facebook::velox::DecodedVector' for 2nd argument
  static int32_t compare(
```